packages/core-react/src/components/Select/index.js

```diff
@@ -10,34 +17,88 @@ export default function Select({
   disabled,
   placeholder,
   label,
-  children
+  active = false,
+  error = false,
+  prepend = false,
+  iconPosition,
+  icon,
+  value = '',
+  children,
+  onChange,
+  ...rest
 }) {
+  const [activeClass, setActiveState] = React.useState(active);
+  const [currValue, setValue] = React.useState(value);
+
+  const handleFocus = () => {
+    setActiveState(true);
+  };
+
+  const handleBlur = () => {
+    setActiveState(false);
+  };
+
+  const handleChange = event => {
+    setValue(event.target.value);
+    onChange(event);
+  };
+
+  React.useEffect(() => {
+    if (value) {
+      setValue(value);
+    }
+  }, []);
+  let iconStart;
+  let iconEnd;
+
+  if (iconPosition) {
+    iconStart = iconPosition === 'iconstart';
+    iconEnd = iconPosition === 'iconend';
+  } else {
+    iconStart = false;
+    iconEnd = false;
+  }
+
   return (
-    <div
-      className={clsx(
-        'ray-select',
-        {
-          'ray-select--compact': compact,
-          'ray-select--disabled': disabled
-        },
-        className
-      )}
-    >
-      <select
-        className="ray-select__input"
-        id={id}
-        name={name}
-        disabled={disabled}
+    <div dir={iconEnd ? 'rtl' : ''}>
+      <div
+        className={clsx(
+          'ray-select',
+          {
+            'ray-select--active': activeClass,
+            'ray-select--has-value': placeholder || currValue,
+            'ray-select--compact': compact,
+            'ray-select--disabled': disabled,
+            'ray-select--error': error,
+            'ray-select--with-prepend': prepend,
+            'ray-select--with-icon-start': iconStart || iconEnd
+          },
+          className
+        )}
       >
-        <option value="" disabled selected data-ray-placeholder>
-          {placeholder}
-        </option>
-        {children}
-      </select>
-
-      <label className="ray-select__label" htmlFor={id}>
-        {label}
-      </label>
+        <InputIcon icon={icon} prepend={prepend} />
+        <div className="ray-select__wrapper">
+          <select
+            className="ray-select__input"
+            name={name}
+            disabled={disabled}
+            onChange={handleChange}
+            onBlur={handleBlur}
+            onFocus={handleFocus}
+            value={currValue}
+            {...rest}
+          >
+            <option value="" disabled selected data-ray-placeholder>
```

Comment:
For the `defaultValue` error i removed the `selected` attribute and that seemed to do the trick! 🤷‍♀